src/include/btree.i

```diff
@@ -1310,20 +1310,40 @@ __wt_page_swap_func(WT_SESSION_IMPL *session, WT_REF *held,
 #endif
 	    );
 
-	/* Expected failures: page not found or restart. */
-	if (ret == WT_NOTFOUND || ret == WT_RESTART)
-		return (ret);
+	/*
+	 * Expected failures: page not found or restart. Our callers list the
+	 * errors they're expecting to handle.
```

Comment:
I think the comment should be clear that the caller is only expecting to handle errors from the `wt_page_in_func` function and making assumptions based on that.  It took me a bit to understand the later code to distinguish these two cases and the user not handling the same error from one of the release functions.